core/src/bundle_stage.rs

```diff
@@ -292,12 +293,25 @@ impl BundleStage {
         bank_start: &BankStart,
         bundle_stage_leader_stats: &mut BundleStageLeaderStats,
         max_bundle_retry_duration: &Duration,
+        reserved_space: &mut u64,
     ) -> BundleStageResult<()> {
         if sanitized_bundle.transactions.is_empty() {
             return Ok(());
         }
 
         let tx_costs = qos_service.compute_transaction_costs(sanitized_bundle.transactions.iter());
+        let total_cost = tx_costs.iter().fold(0, |acc, cost| acc + cost.sum());
+        // If there is enough reserved space left, deduct bundle cost from reserved space before adding
+        if total_cost <= *reserved_space {
+            *reserved_space = *reserved_space - total_cost;
+            let mut dummy_tx_cost = TransactionCost::new_with_capacity(1);
+            dummy_tx_cost.bpf_execution_cost = total_cost;
+            QosService::remove_transaction_costs(
+                vec![dummy_tx_cost].iter(),
+                vec![Ok(())].iter(),
+                &bank_start.working_bank,
+            );
+        }
```

Comment:
There is an edge case, where the OS context switches between these 2 function calls and a tx from a banking stage thread uses the reserved space.  I don't think this will happen very often, but it could be mitigated by pinning this thread to a specific core and reducing it's niceness.

Reply:
at first glance seems fine to me the way it is

Reply:
fixed by holding cost_tracker lock